backend.py

```diff
@@ -853,43 +853,22 @@ def __mkinitrd(mounts, partition, package, kernel_version, fcoe_interfaces):
                              '/etc/init.d/sm-multipath', action]) != 0:
                 raise RuntimeError("Failed to generate multipath configuration")
 
-        # Run mkinitrd inside dom0 chroot
+        # Run dracut inside dom0 chroot
         output_file = os.path.join("/boot", "initrd-%s.img" % kernel_version)
 
         # default to only including host specific kernel modules in initrd
-        if os.path.isdir(os.path.join(mounts['root'], 'etc/dracut.conf.d')):
-            # disable multipath on root partition
-            try:
-                if not isDeviceMapperNode(partition):
-                    f = open(os.path.join(mounts['root'], 'etc/dracut.conf.d/xs_disable_multipath.conf'), 'w')
-                    f.write('omit_dracutmodules+=" multipath "\n')
-                    f.close()
-            except:
-                pass
-        else:
-            args = ['--theme=/usr/share/splash']
-
-            if isDeviceMapperNode(partition):
-                # [multipath-root]: /etc/fstab specifies the rootdev by LABEL so we need this to make sure mkinitrd
-                # picks up the master device and not the slave
-                args.append('--rootdev='+ partition)
-            else:
-                args.append('--without-multipath')
-
-            cmd = ['mkinitrd', '--latch']
-            cmd.extend( args )
-            if util.runCmd2(['chroot', mounts['root']] + cmd) != 0:
-                raise RuntimeError("Failed to latch arguments for initrd.")
-
-        cmd = ['new-kernel-pkg', '--install', '--mkinitrd']
+        # disable multipath on root partition
+        try:
+            if not isDeviceMapperNode(partition):
+                f = open(os.path.join(mounts['root'], 'etc/dracut.conf.d/xs_disable_multipath.conf'), 'w')
+                f.write('omit_dracutmodules+=" multipath "\n')
+                f.close()
+        except:
+            pass
 
-        # Save command used to create initrd in <initrd_filename>.cmd
-        cmd_logfile = os.path.join(mounts['root'], output_file[1:] + '.cmd')
-        cmd_fh = open(cmd_logfile, "w")
-        print(' '.join(cmd + ['"$@"', kernel_version]), file=cmd_fh)
-        cmd_fh.close()
+        cmd = ['dracut', '--no-compress', output_file, kernel_version]
```

Comment:
Why the usage of `--no-compress` ? Currently we compress it

Reply:
no, we do **NOT** compress it in XS8. (stream).
It looks like compress, but actually not, you can check the result in /boot/initrd*.

Not sure whether it is a mkinitrd bug, 
```
        find . -print0 | cpio --null $cpio_owner_root -H newc -o --quiet | \
            $compress >> "${DRACUT_TMPDIR}/initramfs.img";
```

Does NOT generate gzip file while

```
        find . -print0 | cpio --null $cpio_owner_root -H newc -o --quiet | \
            $compress > "${DRACUT_TMPDIR}/initramfs.img";
``` 
does.

Another reasons  we do not compress it explicitly here, is because 
- The compress does not save much disk,   (around 20M?)
- Some xrtuk-xx-xx machine extremely slow to compress, may caused by not fully enable CPU power,
and finally result in installation timeout. (gzip compress takes about 8 minutes)

Reply:
Checked with Lin, current XS9 initrd image is consistent to XS8 that the first cpio party is not compressed and the second party is compressed.

Reply:
@andyhhp  after decompress the result binary, xs9 with `--no-compress` is exactly the same behavior as XS8.

XS8:
```
file initrd-fallback.img 
initrd-fallback.img: ASCII cpio archive (SVR4 with no CRC)

[root@genuk-21-10d ~]# cpio -idv < initrd-fallback.img 
.
early_cpio
kernel
kernel/x86
kernel/x86/microcode
kernel/x86/microcode/AuthenticAMD.bin
kernel/x86/microcode/GenuineIntel.bin
27725 blocks


dd if=initrd-fallback.img of=test.gz bs=512 skip=27725

file test.gz
test.gz: gzip compressed data, from Unix, last modified: Wed Apr 10 06:06:13 2024, max compression
```


XS9:

```
file initrd-fallback.img 
initrd-fallback.img: ASCII cpio archive (SVR4 with no CRC)

 cpio -itv < initrd-fallback.img 
drwxr-xr-x   3 root     root            0 Apr  7 09:19 .
-rw-r--r--   1 root     root            2 Apr  7 09:19 early_cpio
drwxr-xr-x   3 root     root            0 Apr  7 09:19 kernel
drwxr-xr-x   3 root     root            0 Apr  7 09:19 kernel/x86
drwxr-xr-x   2 root     root            0 Apr  7 09:19 kernel/x86/microcode
-rw-r--r--   1 root     root     14117888 Apr  7 09:19 kernel/x86/microcode/GenuineIntel.bin
-rw-r--r--   1 root     root        76166 Apr  7 09:19 kernel/x86/microcode/AuthenticAMD.bin
27725 blocks


 dd if=initrd-fallback.img of=test.gz skip=27725
131398+1 records in
131398+1 records out
67276176 bytes (67 MB, 64 MiB) copied, 0.564197 s, 119 MB/s
[root@genuk-21-09d tmp]# file test.gz
test.gz: gzip compressed data, max compression, from Unix, original size modulo 2^32 214974464
```

Reply:
This is intentional.  The initrd is generally compressed, but microcode is needed so early on boot that it's prepended uncompressed, so it can be accessed directly out of memory.

As a result the initrd appears to be uncompressed, despite most of it being compressed.

Reply:
Can we use `pigz` to make compression faster if it's an issue?

Reply:
I went through the code of dracut and found it will use `pigz` to accelerate and fallback to `gzip -9`. 
I will give it a try.

Reply:
job 3980464, 3980466 shows it can install succeed. (a little bit slower than no-compress)